logos_setup.py

```diff
@@ -0,0 +1,801 @@
+            os.makedirs(os.path.join(HOME, ".config", "Logos_on_Linux"), exist_ok=True)
+            CONFIG_FILE = DEFAULT_CONFIG_PATH
+            if os.path.isdir(os.path.join(HOME, ".config", "Logos_on_Linux")):
+                write_config(CONFIG_FILE, config_keys)
+                logos_info(f"A config file was created at {DEFAULT_CONFIG_PATH}.")
+            else:
+                logos_warn(f"{HOME}/.config/Logos_on_Linux does not exist. Failed to create config file.")
+        elif not CONFIG_FILE and os.path.isfile(DEFAULT_CONFIG_PATH):
+            if logos_acknowledge_question(f"The script found a config file at {DEFAULT_CONFIG_PATH}. Should the script overwrite the existing config?", "The existing config file was not overwritten."):
+                if os.path.isdir(os.path.join(HOME, ".config", "Logos_on_Linux")):
+                    write_config(CONFIG_FILE, config_keys)
+                else:
+                    logos_warn(f"{HOME}/.config/Logos_on_Linux does not exist. Failed to create config file.")
+        else:
+            # Script was run with a config file. Skip modifying the config.
+            pass
+
+        if logos_acknowledge_question("A launch script has been placed in {INSTALLDIR} for your use. The script's name is {FLPRODUCT}.sh.\nDo you want to run it now?\nNOTE: There may be an error on first execution. You can close the error dialog.", "The Script has finished. Exiting…"):
+            subprocess.run([os.path.join(INSTALLDIR, f"{FLPRODUCT}.sh")])
+        elif VERBOSE:
+            print("The script has finished. Exiting…")
+    else:
+        logos_error("Installation failed. {LOGOS_EXE} not found. Exiting…\nThe {FLPRODUCT} executable was not found. This means something went wrong while installing {FLPRODUCT}. Please contact the Logos on Linux community for help.", "")
+
+def install():
+    # BEGIN PREPARATION
+    REGENERATE = os.environ.get('REGENERATE', '')
+    LOGOS_LOG = os.environ.get('LOGOS_LOG')
+    VERBOSE = os.getenv('VERBOSE')
+    if VERBOSE:
+        print(datetime.datetime.now())
+    chooseProduct()  # We ask user for his Faithlife product's name and set variables.
+    if VERBOSE:
+        print(datetime.datetime.now())
+    chooseVersion()  # We ask user for his Faithlife product's version, set variables, and create project skeleton.
+    if VERBOSE:
+        print(datetime.datetime.now())
+    getLogosReleaseVersion(getLogosReleases())
+    if VERBOSE:
+        print(datetime.datetime.now())
+    logos_setup() # We set some basic variables for the install, including retrieving the product's latest release.
+    if VERBOSE:
+        print(datetime.datetime.now())
+    chooseInstallMethod()  # We ask user for his desired install method.
+    # END PREPARATION
+    
+    TARGETVERSION = os.environ.get('TARGETVERSION')
+    LOGOS_EXE = os.environ.get('LOGOS_EXE')
+    FLPRODUCT = os.environ.get('FLPRODUCT')
+    WINEPREFIX = os.environ.get('WINEPREFIX')
+
+    if REGENERATE == "":
```

Comment:
This var is related to the launcher and controlpanel scripts and can be deleted. We'll need to remove it from the rest of the project, or determine if it needs to be repurposed under python.

Reply:
I've removed this in a "patch" function I made, `logos_setup.finish_install()`, but I've left it in the main `logos_setup.install()` function for now. (I hope to eventually eliminate my patch function, but I need it for the Tk window for now because the steps are broken up a little differently than in the curses menu.)